plugins/coverbrowser.koplugin/main.lua

```diff
@@ -509,14 +509,63 @@ function CoverBrowser.initGrid(menu, display_mode)
     menu.display_mode_type = display_mode and display_mode:gsub("_.*", "") -- "mosaic" or "list"
 end
 
+function CoverBrowser.addFileDialogButtons(widget)
+    FileManager.addFileDialogButtons(widget, "coverbrowser_1", function(file, is_file, bookinfo)
```

Comment:
(It's good these use `.` and not `:` as they are utilities _functions_. Took me a few seconds to get that right - widget can be a `FileManager` instance :) and we will call `FileManager` (class) `.addFileDialogButtons( FileManager` (instance) `)` -  wondering if it needs a comment - or nevermind.)

Reply:
To add buttons, in an outer module we call:
`FileManager:addFileDialogButtons(row_id, row_func)`

(In CoverBrowser we do `FileManager.addFileDialogButtons(FileManager, row_id, row_func)` for unification)

`FileManager.addFileDialogButtons(FileManagerHistory, row_id, row_func)`

`FileManager.addFileDialogButtons(FileManagerCollections, row_id, row_func)`

Reply:
To avoid triplicating the code.

Reply:
> and we will call `FileManager` (class) `.addFileDialogButtons( FileManager` (instance) `)`

No, additional buttons are saved to the baseclass.

Reply:
For FM we can add them to FileChooser, that was a choice.
I decided that FileManager was more logical, since the additional buttons to be added to the FM file dialog.

Reply:
I know, I did understand all this and it is perfect - explanations were not needed :)

It's just that reading `FileManager.addFileDialogButtons(FileManager, row_id, row_func)` felt odd, until I noticed that the first `.` wasn't a `:` and that `FileManager.addFileDialogButtons` is a function and not a method. Some little comment may have saved me a few seconds :)